### PR TITLE
Qute: add basic date-time formatting support

### LINKED ISSUE #11181: Date and Time Formatting for Qute
**Description**
As discussed in the Zulip [chat](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Qute.20Date.2FTime.20Format), it would be nice if Qute supported formatting of date/time strings if a date or time object is passed into the template renderer.  An example might look lik this:

```
{time:format(myTime,'d MMM uuuu')}
```

**Implementation ideas**
It was suggested in the chat discussion that a template extension would be the way to achieve this.

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #11181

### CODE REVIEW

**Comment by mkouba** [comment]
CC @aelfric

File: extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/extensions/TimeTemplateExtensionsTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                            "templates/foo.html"));
+
+    @Inject
+    Template foo;
+
+    @Test
+    public void testFormat() {
+        Calendar nowCal = Calendar.getInstance();
+        nowCal.set(Calendar.YEAR, 2020);
+        nowCal.set(Calendar.MONTH, 8);
```
Comment (gastaldi) [inline extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/extensions/TimeTemplateExtensionsTest.java:38]:
```suggestion
        nowCal.set(Calendar.MONTH, Calendar.SEPTEMBER);
```

**Review by gastaldi — APPROVED** [review]
LGTM! Added a minor suggestion (just for clarification purposes, doesn't change anything really)

**Comment by gastaldi** [comment]
@mkouba thanks for merging my suggestion. Can you squash all commits before merging?

**Pull request merged**